src/middleware/devtools.ts

```diff
@@ -1,167 +1,325 @@
+    }
+  | { type: UnknownString }
+
+type EStateLifted =
+  { computedStates: { state: O.Partial<EStateFromDevtools> }[] }
+type EDevtoolsPrefix = string & { __isDevtoolsPrefix: true }
+type EActionToSend = { type: string } & { __isActionToSend: true }  
+type EStateFromDevtools = { __isStateFromDevtools: true }
+
+const devtoolsWindow = window as
+  & Window
+  & DevtoolsWindow
+  & { top: DevtoolsWindow }
+
+const devtoolsImpl: EDevtools =
+  (storeInitializer, _devtoolsOptions) =>
+    (parentSet, parentGet, parentStore) => {
+
+  let devtoolsOptions: EDevtoolsOptions =
+    _devtoolsOptions === undefined ? { name: undefined } :
+    typeof _devtoolsOptions === 'string' ? { name: _devtoolsOptions } :
+    _devtoolsOptions
+
+  let extensionConnector =
+    devtoolsWindow.__REDUX_DEVTOOLS_EXTENSION__ ??
+    devtoolsWindow.top.__REDUX_DEVTOOLS_EXTENSION__
+
+  if (!extensionConnector) {
+    if (process.env.NODE_ENV === 'development' && typeof window !== 'undefined') {
+      console.warn('Please install/enable Redux devtools extension')
+    }
+
+    return storeInitializer(parentSet, parentGet, parentStore)
+  }
+
+  let store: EStore & EDevtoolsStore = parentStore;
+  if (!store.devtools) store.devtools = extensionConnector.connect(devtoolsOptions);
+  store.devtools.prefix = prefixFromName(devtoolsOptions.name);
+
+  let isSilentSetState = false;
+  store.setState = (...[state, replace, action]: F.Arguments<typeof store['setState']>) => {
+    parentSet(state, replace)
+    if (!isSilentSetState) return;
+    store.devtools?.send(
+      actionToSend(action, store.devtools.prefix),
+      parentGet()
+    )
+  }
+  const silentlySetState = (state: EStateFromDevtools | O.Partial<EStateFromDevtools>) => {
+    isSilentSetState = true;
+    store.setState(state as unknown as EState);
+    isSilentSetState = false;
+  }
+
+  let initialState =
+    storeInitializer(store.setState, parentGet, store);
+
+  store.devtools.subscribe(message => {
+    switch (message.type) {
+      case 'ACTION':
+        if (!isReduxLike(store)) return;
+        if (typeof message.payload !== "string") return;
+        return parseJsonThen(message.payload, store.dispatch);
+
+      case 'DISPATCH':
+        if (!store.devtools) return;
+        switch (message.payload.type) {
+          case 'RESET': return store.devtools.init(initialState);
+          case 'COMMIT': return store.devtools.init(store.getState());
+          case 'ROLLBACK': return parseJsonThen(message.state, store.devtools.init);
+          case 'JUMP_TO_STATE':
+          case 'JUMP_TO_ACTION':
+            return parseJsonThen(message.state, silentlySetState);
+          case 'IMPORT_STATE':
+            let { nextLiftedState } = message.payload;
+            let lastComputedState = nextLiftedState.computedStates.at(-1)?.state
+            if (!lastComputedState) return;
+            silentlySetState(lastComputedState);
+            store.devtools.send(null, nextLiftedState);
```

Comment:
Previously we did something else, which I'm not sure is entirely correct, so this new thing is from [here](https://github.com/zalmoxisus/mobx-remotedev/blob/2d9ea51d3378e0507cfe8295046491e9f25c4a77/src/monitorActions.js#L77-L80).

Reply:
This should also be in a separate PR, if it fixes a behavior.
Otherwise, we will not be able to review the PR.